client-side/src/components/profileComponents/updateProfileCpmponent.jsx

```diff
@@ -0,0 +1,408 @@
+import React, { useState, useEffect } from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Alert from '@mui/material/Alert';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import GenericInput from '../../stories/GenericInput/genericInput.jsx';
+import Select from '../../stories/Select/Select.jsx';
+import GenericButton from '../../stories/Button/GenericButton.jsx';
+import { updateProfileApi, deleteProfileApi } from '../../services/profileService.js';
+import { createWebsite, updateWebsite } from '../../services/websiteService.js';
+import { deleteProfile } from '../../redux/profile/profile.slice.js';
+import { useDispatch } from 'react-redux';
+import ToastMessage from '../../stories/Toast/ToastMessage.jsx';
+export default function UpdateProfileComponent({ profile, onProfileUpdated }) {
+    const dispatch = useDispatch();
+    const [open, setOpen] = useState(false);
+    const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+    const [toastOpen, setToastOpen] = useState(false);
```

Comment:
as I already wrote in addProfile.jsx, the toast messages management shouldnt be here but be common to he whole app. please check what is the conventional way and fix it.